File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectWoonplaats.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;


/**
 * ZaakobjectWoonplaats
 */
public class ZaakobjectWoonplaats extends ZaakobjectMetObjectIdentificatie<ObjectWoonplaats> {

    /**
     * Constructor for JSONB deserialization
     */
    public ZaakobjectWoonplaats() {
    }

    /**
     * Constructor with required attributes
     */
    public ZaakobjectWoonplaats(final URI zaak, final URI bagobjectUri, final ObjectWoonplaats woonplaats) {
        super(zaak, bagobjectUri, ObjectTypeEnum.WOONPLAATS, woonplaats);
    }

    @Override
    public String getWaarde() {
        return getObjectIdentificatie().getIdentificatie();
    }

}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectWoonplaats.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI
import java.util.UUID

class ZaakobjectWoonplaats : ZaakobjectMetObjectIdentificatie<ObjectWoonplaats> {
    /**
     * Constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with required attributes
     */
    constructor(
        zaak: URI,
        bagobjectUri: URI?,
        woonplaats: ObjectWoonplaats?,
        url: URI,
        uuid: UUID
    ) : super(zaak, bagobjectUri, ObjectTypeEnum.WOONPLAATS, woonplaats, url, uuid)

    override val waarde: String?
        get() = objectIdentificatie?.identificatie
}
